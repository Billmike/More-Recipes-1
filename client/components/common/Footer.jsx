import React from 'react';
import { Link } from 'react-router-dom';

/**
 * Footer component
 *
 * @returns {object} React element
 */
const Footer = () => (
  <div className="page-footer deep-orange darken-1">
    <div className="container center-align">
      <div className="row">
        <div className="connect-icons col s12 m6 l6">
          <h5 className="white-text">Stay Connected</h5>
          <i className="fa fa-facebook-square fa-2x" />
          <i className="fa fa-twitter-square fa-2x" />
          <i className="fa fa-pinterest-square fa-2x" />
          <i className="fa fa-instagram fa-2x" />
        </div>
        <div className="col s12 m6 l4 offset-l2">
          <h5 className="white-text">Navigate</h5>
          <ul>
            <li>
              <Link to="/" className="grey-text text-lighten-3">Home</Link>
            </li>
            <li>
              <a className="grey-text text-lighten-3" href="#!">About Us</a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© 2017 All rights reserved
      </div>
    </div>
  </div>
);

export default Footer;
