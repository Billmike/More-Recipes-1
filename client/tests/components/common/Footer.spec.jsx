import React from 'react';

import Footer from '../../../components/common/Footer';

describe('Footer component', () => {
  it('should renders footer', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('div').length).toBe(7);
    expect(wrapper.find('h5').length).toBe(2);
    expect(wrapper.find('i').length).toBe(4);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('li').length).toBe(2);
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
