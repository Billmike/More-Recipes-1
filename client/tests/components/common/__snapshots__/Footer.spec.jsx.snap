// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should renders footer 1`] = `
<div
  className="page-footer deep-orange darken-1"
>
  <div
    className="container center-align"
  >
    <div
      className="row"
    >
      <div
        className="connect-icons col s12 m6 l6"
      >
        <h5
          className="white-text"
        >
          Stay Connected
        </h5>
        <i
          className="fa fa-facebook-square fa-2x"
        />
        <i
          className="fa fa-twitter-square fa-2x"
        />
        <i
          className="fa fa-pinterest-square fa-2x"
        />
        <i
          className="fa fa-instagram fa-2x"
        />
      </div>
      <div
        className="col s12 m6 l4 offset-l2"
      >
        <h5
          className="white-text"
        >
          Navigate
        </h5>
        <ul>
          <li>
            <Link
              className="grey-text text-lighten-3"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#!"
            >
              About Us
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="footer-copyright"
  >
    <div
      className="container"
    >
      Â© 2017 All rights reserved
    </div>
  </div>
</div>
`;
